common/djangoapps/status/status.py

```diff
@@ -4,48 +4,34 @@
 
 from django.conf import settings
 from django.core.cache import cache
-import json
 import logging
-import os
+
+from .models import GlobalStatusMessage
 
 log = logging.getLogger(__name__)
 
 
-def get_site_status_msg(course_id):
+def get_site_status_msg(course_key):
     """
-    Look for a file settings.STATUS_MESSAGE_PATH.  If found, read it,
-    parse as json, and do the following:
-
-    * if there is a key 'global', include that in the result list.
-    * if course is not None, and there is a key for course.id, add that to the result list.
-    * return "<br/>".join(result)
+    Pull the status message from the database.
 
-    Otherwise, return None.
-
-    If something goes wrong, returns None.  ("is there a status msg?" logic is
-    not allowed to break the entire site).
+    Caches the message by course.
     """
     try:
-        # first check for msg in cache
-        msg = cache.get('site_status_msg')
-        if msg is not None:
-            return msg
-
-        if os.path.isfile(settings.STATUS_MESSAGE_PATH):
-            with open(settings.STATUS_MESSAGE_PATH) as f:
-                content = f.read()
-        else:
+        # The current() value for GlobalStatusMessage is cached.
+        if not GlobalStatusMessage.current().enabled:
             return None
 
-        status_dict = json.loads(content)
-        msg = status_dict.get('global', None)
-        if course_id in status_dict:
-            msg = msg + "<br>" if msg else ''
-            msg += status_dict[course_id]
+        cache_key = "status_message.{course_id}".format(course_id=unicode(course_key))
+        if cache.get(cache_key):
+            return cache.get(cache_key)
+
+        course_status_message = GlobalStatusMessage.current().full_message(course_key)
+        cache.set(cache_key, course_status_message)
```

Comment:
Wouldn't it be better to build this caching logic into the `full_message` method?